plugin/restore/mysql/mysql.go

```diff
@@ -235,10 +235,13 @@ func getBinlogReplayList(startBinlogInfo api.BinlogInfo, binlogDir string) ([]st
 		return nil, err
 	}
 
-	for i := 1; i < len(binlogFilesToReplaySorted); i++ {
-		if binlogFilesToReplaySorted[i].Seq != binlogFilesToReplaySorted[i-1].Seq+1 {
-			return nil, fmt.Errorf("discontinuous binlog file extensions detected in %s and %s", binlogFilesToReplaySorted[i-1].Name, binlogFilesToReplaySorted[i].Name)
-		}
+	if binlogFilesToReplaySorted[0].Seq != startBinlogSeq {
```

Comment:
Is it possible to have len(binlogFilesToReplaySorted) == 0?

Reply:
Yes, and I find the current implementation deals with the "no binlog files found" error wrong.
Fixed.